### PR TITLE
Fixes documentation in a lot of places. Fixes #51.

### LINKED ISSUE #51: Cross-reference and code highlight documentation
We should add cross-references in the haddock, e.g. writing `'MSF'` in most places instead of `MSF` or `@MSF@`. Where no useful reference exist, still prefer monospace `@someHaskellConcept@` over someHaskellConcept.

### PR DESCRIPTION
**Pull request by turion**

### CODE REVIEW

File: src/Control/Monad/Trans/MSF/Reader.hs [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
--- | MSFs with a Reader monadic layer.
+-- | 'MSF's with a 'Reader' monadic layer.
 --
--- This module contains functions to work with MSFs that include a 'Reader'
--- monadic layer. This includes functions to create new MSFs that include an
--- additional layer, and functions to flatten that layer out of the MSF's
+-- This module contains functions to work with 'MSF's that include a 'Reader'
+-- monadic layer. This includes functions to create new 'MSF's that include an
+-- additional layer, and functions to flatten that layer out of the 'MSF''s
```
Comment (ivanperez-keera) [inline src/Control/Monad/Trans/MSF/Reader.hs]: Not a big deal, but this will render incorrectly (the double single quote).

**Comment by turion** [comment]
Do you know how to format the ' correctly?

**Comment by ivanperez-keera** [comment]
According to https://github.com/haskell/haddock/issues/162, you need to write it like this: ```'MSF`'s```.

This happens in three modules (`Reader`, `State` and `Writer`).

**Comment by turion** [comment]
Thanks for the hint!

**Comment by turion** [comment]
Merging since the remaining minor formatting issue is taken care of.

**Pull request merged**